llvm/lib/Target/AMDGPU/GCNSchedStrategy.cpp

```diff
@@ -63,6 +63,10 @@ static cl::opt<bool> GCNTrackers(
     cl::desc("Use the AMDGPU specific RPTrackers during scheduling"),
     cl::init(false));
 
+static cl::opt<bool> UseAMDGPUScheduleHeuristic(
+    "amdgpu-use-amdgpu-schedule-heuristic", cl::Hidden,
```

Comment:
Name and description too general

Also it should be on by default, if an option at all

Reply:
I think it is OK to be default on (the only difference would be lots of test changes). But I would like we could keep it off for the first submission (at least before we see some performance results on compute benchmarks). We may still need fine-tuning the heuristics after more extensive testing. We can turn it on later once this version's tryCandidate() becomes somewhat stable and do not regress much real world cases. I am not sure how such kind of performance testing is done in compute side. It would better we can get some performance numbers for the change. Please let me know if you still have concern on this.


> Name and description too general

Do you have any suggestion? Maybe `amdgpu-use-custom-try-candidate`?

Reply:
Some of the scheduling changes I'm working on involve having multiple selectable versions of tryCandidate. I think we can assume that future work on the scheduler will always include a custom tryCandidate override. We also already override it in the max-ilp scheduler. So I agree, the name is too broad. I think you should make this a separate SchedStrategy and have it be selectable that way.

Reply:
Thanks @kerbowa for the suggestion! I have make it a separate strategy.